bundle.go

```diff
@@ -181,28 +205,37 @@ func (b ValidateBundles) hasDuplicatedVersions() bool {
 // version bundles. Lists of version bundles reflect distributions.
 type ValidateAggregatedBundles [][]Bundle
 
-// TODO add tests for deprecated bundles within a group
 func (b ValidateAggregatedBundles) Validate() error {
 	if len(b) != 0 {
 		l := len(b[0])
 		for _, group := range b {
 			if l != len(group) {
-				return microerror.Mask(invalidBundleError)
+				return microerror.Maskf(invalidAggregatedBundlesError, "number of version bundles within aggregated version bundles must be qual")
 			}
 		}
 	}
 
-	for _, group := range b {
-		var deprecatedCount int
-		for _, bundle := range group {
-			if bundle.Deprecated {
-				deprecatedCount++
+	if b.hasDuplicatedAggregatedBundles() {
+		return microerror.Maskf(invalidAggregatedBundlesError, "version bundles within aggregated version bundles must be unique")
+	}
+
+	return nil
+}
+
+func (b ValidateAggregatedBundles) hasDuplicatedAggregatedBundles() bool {
+	for _, b1 := range b {
+		var seen int
+
+		for _, b2 := range b {
+			if reflect.DeepEqual(b1, b2) {
+				seen++
+
+				if seen >= 2 {
```

Comment:
Why 2?

Reply:
We iterate over the same list twice. This means we will see each item at least once. In case we see it twice it is duplicated in the list.

Reply:
Ack.